src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -205,6 +209,111 @@ public void writeVariableExpr_scopedStaticFinalDecl() {
     assertEquals(writerVisitor.write(), "public static final boolean x");
   }
 
+  @Test
+  public void writeNormalBlockComment() {
+    String content = "this is a test comment";
+    BlockComment blockComment = BlockComment.builder().setComment(content).build();
+    String expected = "/** this is a test comment */\n";
+    blockComment.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), expected);
+  }
+
+  @Test
+  public void writeBadBlockComment() {
+    String content =
+        "A super long long long long long long long */ long long long long long long long comment";
```

Comment:
I would suggest adding the escaper in a separate PR once the Escaper in string object value is merged.
1. comment is still a string, so the escaper for comments would have the same replacement that StringObjectValue currently has, but not only those.

2. as the test case shown above, we need extra replacement to handle unexpected delimiter `*/` in block/javaDocComment, `//` won't be a problem.

so I think we can have another `escapeComment` method calling the escaper in JavaEscaper class from [here](https://github.com/googleapis/gapic-generator-java/pull/72/files#diff-8d3414d270a69cc5189e4da37556b8f3R40).

And this test should stay in JavaWriterVisitorTest, because we perform the special handling in JavaWriterVisitor. The comment classes only collect the comment components, not formatting them.

Reply:
This escaper should be separate from the one in the other PR, and can probably be a class member of BlockComment. As suggested above, this particular escaping should be part of BlockComment because it concerns the comment contents, and not the `/**` or `//` parts.

Reply:
I see, this sounds reasonable to me, as you suggested below, I think a comment escaper in `escaper` folder and calling it in BlockComment/JavaDocComment classes to handle the comment contents is a great way.